### PR TITLE
feat: support node public IP tags

### PR DESCRIPTION
**Pull request by nilo19**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az aks create --node-public-ip-tags
az aks nodepool add --node-public-ip-tags

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -6138,3 +6138,63 @@ def test_aks_create_and_update_ssh_public_key(self, resource_group, resource_gro
         # delete
         self.cmd(
             'aks delete -g {resource_group} -n {name} --yes --no-wait', checks=[self.is_empty()])
+
+    @AllowLargeResponse()
+    @AKSCustomResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='eastus2euap')
+    def test_node_public_ip_tags(self, resource_group, resource_group_location):
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:6144]: I [tested](https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=63836715&view=results) the case but it failed on feature not registered. Could you please add custom headers to the case?

Comment (nilo19) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:6144]: done

**Review by FumingZhang — APPROVED** [review]
LGTM, [test passed](https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=63843273&view=results).

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**